src/Util/Logger/WorkerLogger.php

```diff
@@ -37,11 +37,11 @@ public function __construct(LoggerInterface $logger, $functionName)
 	 * Generates an UID
 	 *
 	 * @param $length
-	 * @return bool|string
+	 * @return string
 	 */
 	private function generateUid($length)
 	{
-		return substr(hash('md5', uniqid('', true)), 0, $length);
+		return bin2hex(random_bytes($length / 2));
```

Comment:
I appreciate you taking my exact code, but I really can't predict the behavior of this code for odd lengths.

Reply:
You're right, `$lenght = 7` shortens the uid to 6 characters f.e. ..I'll use another way..

Reply:
I'm now using `uniqid()` in combination with `substr()` but without `hash()` again.. The tests are working with different uid lengths and the output seems right.